### DISCUSSION TITLE
Updating the container element via setOptions seamingly does nothing

### DISCUSSION THREAD

**Discussion by anton7r** [katspaugh/wavesurfer.js#3197] [Q&A]
## Details

React `v18.2.0`
Wavesurfer.js `v7.3.2`

## Background

I have created a virtualized list component that has to be able to "display" thousands of rows each of which have one wavesurfer instance created only when they are visible. Once the wavesurfer instance is created it is then stored in a global state in order to avoid recreating the wavesurfer instance when it becomes visible again as creating the wavesurfer instance and drawing the waveform from scratch is somewhat expensive.

The main reason why im currently storing the wavesurfer instance in a global store is to prevent the application from loading the audio assets multiple times (the issue could be solved also with just storing the audio assets as `Blobs` in the global store). The other not so important reason is to prevent recalculation and redrawing of the waveform.

Im open to sharing more about the current implementation as the project itself is going to be open source in the future. The repository is currently private since the project is still early in it's development.

## The issue

Currently there is an issue while trying to remount the wavesurfer instance to a new container element set via the `setOptions` method.
As it seems updating the container element via `setOptions` seamingly does nothing.
```ts
// Variables in use
const wavesurfer = Wavesurfer.create({...})
const mediaEl: HTMLMediaElement = ...;
const containerEl: HTMLElement = ...;

// The code itself
wavesurfer.setOptions({
  media: mediaEl,
  container: containerEl,
});
```

## Related discussions
#3191 Solves the recomputation and redrawing issue via caching in local storage.

**Comment by anton7r** [comment]
The solution discussed in #3191 works like a charm.

But the uncertainty still remains whether or not updating the container via `setOptions` should work or not.

**Reply by katspaugh** [reply]
setOptions intentionally supports only a subset of parameters, mostly things like colors. The argument type could be probably more informative, my bad.

I’m open to adding container handling there if you want to have it. Feel free to make a PR for it.

**Reply by anton7r** [reply]
Have been working on a similar solution as in #3191, but it does not completely work at the moment.

I'd find adding the ability to set the container to be simpler therefore i'll be making a pr for it.

**Reply by anton7r** [reply]
Pr opened #3208
